gse-copypaste/gse-copypaste-afs/src/main/java/com/powsybl/gse/copypaste/afs/CopyManager.java

```diff
@@ -0,0 +1,226 @@
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+/**
+ * @author Nassirou Nambiema <nassirou.nambiena at rte-france.com>
+ */
+public final class CopyManager {
+
+    private final String storageDirectory;
+
+    private final Object copyLock = new Object();
+    private ExecutorService tPool = Executors.newFixedThreadPool(3);
+    private Map<String, CopyInfo> copy = new HashMap<>();
+
+    public CopyManager() {
+        initCopy();
+        storageDirectory = getStorageDirectory();
+    }
+
+    /**
+     * @param nodes the nodes to copy
+     */
+    public Map<String, CopyInfo> copy(List<? extends AbstractNodeBase> nodes) {
+
+        for (AbstractNodeBase node : nodes) {
+            CopyInfo copyInfo = new CopyInfo();
+            copyInfo.nodeId = node.getId();
+            copyInfo.node = node;
+            copyInfo.archivePath = Paths.get(storageDirectory + node.getName() + node.getId());
+            copyInfo.archiveComplete = false;
+            copyInfo.expirationDate = ZonedDateTime.now().plusHours(CopyServiceConstants.COPY_EXPIRATION_TIME);
+            tPool.execute(() -> {
+                // create task
+                try {
+                    archiveAndCopy(node);
+                } catch (Exception e) {
+                    throw new PowsyblException("The node " + node.getName() + " copy have encountered some problems! ");
+                }
+                // change completion status + end task
+                copyInfo.archiveComplete = true;
+            });
+            copy.put(copyInfo.nodeId, copyInfo);
+        }
+        return copy;
+    }
+
+    /**
+     * @param nodeId the copied node's id
+     * @param folder the destination's folder
+     */
+    public void paste(String nodeId, AbstractNodeBase folder) {
+        throwNodeIsNotAFolderException(folder);
+
+        boolean nodeNameAlreadyExists = false;
+        CopyInfo copyInfo = copy.get(nodeId);
+
+        if (copyInfo != null && copyInfo.archiveComplete) {
+            if (folder instanceof ProjectFolder) {
+                List<ProjectNode> children = ((ProjectFolder) folder).getChildren();
+                nodeNameAlreadyExists = children.stream().anyMatch(child -> copyInfo.node.getName().equals(child.getName()));
+            } else if (folder instanceof Folder) {
+                List<Node> children = ((Folder) folder).getChildren();
+                nodeNameAlreadyExists = children.stream().anyMatch(child -> copyInfo.node.getName().equals(child.getName()));
+            }
+            if (nodeNameAlreadyExists) {
+                renameAndPaste(folder, copyInfo);
+            } else {
+                folder.unarchive(copyInfo.archivePath);
+            }
+        }
+    }
+
+    private void initCopy() {
```

Comment:
you can just call it "init", it is otherwise confusing since one can think it initialize a copy.